identity-core/src/crypto/signature/errors.rs

```diff
@@ -0,0 +1,219 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+//! Signature related errors used in this crate
+
+use thiserror::Error as DeriveError;
+
+use crate::crypto::key::KeyError;
+
+pub use signing::MissingSignatureError;
+pub use signing::SigningError;
+pub(crate) use signing::SigningErrorCause;
+pub use verifying::ProofValueError;
+pub use verifying::VerificationError;
+pub use verifying::VerificationProcessingError;
+
+mod signing {
+  use crate::convert::JsonDecodingError;
+  use crate::convert::JsonEncodingError;
+
+  use super::DeriveError;
+  use super::KeyError;
+  #[derive(Debug, DeriveError)]
+  /// Caused by a failed attempt at retrieving a digital signature.
+  #[error("signature not found")]
+  pub struct MissingSignatureError;
+
+  #[derive(Debug, DeriveError)]
+  // The reason why a signing operation failed.
+  pub(crate) enum SigningErrorCause {
+    // A signing operation failed because the signature could not be set
+    #[error("signing failed - unable to access the required signature: {0}")]
+    MissingAccess(&'static str),
+    // Signing failed because the signing method received invalid input
+    #[error("signing failed - invalid input:  {0}")]
+    Input(&'static str),
+    // Any reason why signing failed that is not necessarily listed here
+    #[error("signing failed: {0}")]
+    Other(&'static str),
+  }
```

Comment:
Even when the type is `pub(crate)` ?

Reply:
Missed that when I wrote it. It's less important then, but since it wouldn't show up in documentation on `docs.rs` I think it would be fine. Makes me wonder if we should then also change our recommended way of building the docs in the top-level README to not include `--document-private-items`. Overall, I'm fine either way.